packages/form-field-scripts/src/form-field-webpack-plugin/verifications.js

```diff
@@ -0,0 +1,159 @@
+const require = createRequire(import.meta.url)
+
+const performOnetimeChecks = async () => {
+    // These validations are only performed ones (at instantiating time),
+    // if the dev tries to change things that requires re-instantiating, for example, if he modifies the form-field.config.js or package.json,
+    // the build become obsolete and needs to be restarted again.
+    // Add a file watcher that watch the form-field.config.js and package.json of the custom form field project from which this Webpack plugin is getting instantiating,
+    // kill the build if changes are detecting in these files (add a log explaining why the build was killed, so that the user doesn't get
+    // confused)...
+    await verifyPackageVersions()
+    await validateFormFieldProjectConfigSchema()
+    await isAllMandatoryModulesPresent()
+}
+
+const performRuntimeChecks = async () => {
+    await defaultExportCheckAllModules()
+}
+
+const performPreBuildtimeChecks = async () => {
+    // No need to check if package.json file exists or not, because if package.json doesn't exists the scripts that run the current file
+    // can't won't be invoked.
+    await getFormFieldProjectConfig() // needs need to move 1 level above the compilation tree.
+}
+
+const defaultExportCheckAllModules = async () => {
+    const modulesPresentInFormFieldProject =
+        await getComponentsFromFormFieldProject()
+
+    for (let [platforms, components] of Object.entries(
+        modulesPresentInFormFieldProject
+    )) {
+        for (const [component, modulePath] of Object.entries(components)) {
+            let hasDefaultExport = false
+            let isReactComponent = false
+            let code = ''
+
+            try {
+                code = fs.readFileSync(
+                    path.join(paths.appRoot, modulePath),
+                    'utf-8'
+                )
+            } catch (err) {
+                throw new FORM_FIELD_MODULE_NOT_PRESENT_ERROR({ modulePath })
+            }
+
+            try {
+                const ast = babel.parse(code, {
+                    presets: [
+                        require('@babel/preset-env'),
+                        require('@babel/preset-react'),
+                    ],
+                })
+
+                babel.traverse(ast, {
+                    ExportDefaultDeclaration(path) {
+                        hasDefaultExport = true
+                    },
+                })
+            } catch (err) {
+                throw new UNABLE_TO_PARSE_A_FORM_FIELD_MODULE({
+                    component,
+                    modulePath,
+                    err,
+                })
+            }
+
+            if (!hasDefaultExport) {
+                throw new DEFAULT_EXPORT_NOT_FOUND_ERROR({
+                    component,
+                    modulePath,
+                })
+            }
+        }
+    }
+}
+
+const validateFormFieldProjectConfigSchema = async () => {
+    const ajv = new Ajv()
```

Comment:
json schema validator package is same as satheesh used in json variable ?

Reply:
This is ajv, the json schema validation that is getting performed here has nothing to do with what we are doing in our webapp (kf-xg-frontend), this schema validation is present to validate the structure of form-field.config.js...